Firestore/Source/API/FIRQuerySnapshot.mm

```diff
@@ -123,15 +123,15 @@ - (NSInteger)count {
 
 - (NSArray<FIRDocumentChange *> *)documentChangesWithIncludeMetadataChanges:
     (BOOL)includeMetadataChanges {
-  if (includeMetadataChanges && _snapshot->view_snapshot().excludes_metadata_changes()) {
-    ThrowInvalidArgument("To include metadata changes with your document changes, you must call "
-                         "addSnapshotListener(includeMetadataChanges: true).");
-  }
-
   if (!_documentChanges || _documentChangesIncludeMetadataChanges != includeMetadataChanges) {
-    _documentChanges = [FIRDocumentChange documentChangesForSnapshot:_snapshot->view_snapshot()
-                                              includeMetadataChanges:includeMetadataChanges
-                                                           firestore:_snapshot->firestore()];
+    NSMutableArray *documentChanges = [NSMutableArray array];
+    _snapshot->ForEachChange(
+        includeMetadataChanges == YES, [&documentChanges](DocumentChange change) {
```

Comment:
Nit: Gil was recently purging existing usages of `YES`/`NO`, maybe replace with `static_cast<bool>(includeMetadataChanges)`?